src/model.py

```diff
@@ -151,6 +153,14 @@ def init_engine(self):
             AsyncEngineArgs(**self.vllm_engine_config)
         )
 
+        # Create vLLM custom Metrics
+        labels = {
+            "model": self.args["model_name"],
+            "version": self.args["model_version"],
+        }
+        logger = VllmStatLogger(labels=labels)
+        self.llm_engine.add_logger("triton", logger)
```

Comment:
What is the cadence that the logger gets called? CC @kthui as this will involve round trips with core, similar to your investigation with request cancellation frequency.

Reply:
Can you elaborate on this?

Reply:
How often will the metrics get updated? Every request, every token, every full response, etc. ? in other words, how often will vLLM engine call this attached triton stats logger?

Reply:
every iteration

Reply:
That will probably significantly affect the total throughput then if the core round trip communication will interrupt the generation at every iteration based on Jacky and Iman's recent findings. We probably want this feature either way - just calling out that we'll likely need to make similar optimizations that @kthui is doing right now for this feature.

Reply:
@kthui will run benchmarks.

Reply:
The current path forward is to allow metrics to be turned off. There is still room to improve in the future, i.e. perform the core round-trip communication on a side branch.

Reply:
At this point, the impact of having metrics (counter ***and gauge***) on performance with `--disable-log-stats` flag set on FastAPI completion vs Triton generate_stream is negligible. The delta between FastAPI completion and Triton generate_stream without any metrics functionality added is approximately the same with metrics and having the `--disable-log-stats` flag set.